overlord/assertstate/assertstate.go

```diff
@@ -356,15 +356,52 @@ func RefreshValidationSetAssertions(s *state.State, userID int) error {
 		return nil
 	}
 
-	if err := bulkRefreshValidationSetAsserts(s, vsets, userID, deviceCtx); err != nil {
-		return err
+	monitorModeSets := make(map[string]*ValidationSetTracking)
+	enforceModeSets := make(map[string]*ValidationSetTracking)
+	for vk, vset := range vsets {
+		if vset.Mode == Monitor {
+			monitorModeSets[vk] = vset
+		} else {
+			enforceModeSets[vk] = vset
+		}
+	}
+
+	monitorModeRefreshErr := bulkRefreshValidationSetAsserts(s, monitorModeSets, nil, userID, deviceCtx)
+
+	checkForConflicts := func(db *asserts.Database, bs asserts.Backstore) error {
+		vsets := snapasserts.NewValidationSets()
+		tmpDb := db.WithStackedBackstore(bs)
+		for _, vs := range enforceModeSets {
+			headers := map[string]string{
+				"series":     release.Series,
+				"account-id": vs.AccountID,
+				"name":       vs.Name,
+			}
+			as, err := tmpDb.FindSequence(asserts.ValidationSetType, headers, -1, asserts.ValidationSetType.MaxSupportedFormat())
```

Comment:
we never remove assertions, so there might be in principle independently added/left over assertions so I think we need to use the right sequence number and not search for the latest if the validation set is currently pinned